SMDS_3/include/CGAL/Simplicial_mesh_cell_base_3.h

```diff
@@ -32,187 +32,48 @@
 
 namespace CGAL {
 
-// Class Simplicial_mesh_cell_3
-// Cell base class used for tetrahedral remeshing
-// Adds information to Cb about the cell of the input complex containing it
-template< class Subdomain_index_,
-          class Surface_patch_index_,
-          class TDS>
+template <typename Gt,
+          typename SubdomainIndex,
+          typename SurfacePatchIndex,
+          typename Cb>
 class Simplicial_mesh_cell_3
+  : public Cb
 {
 public:
-  using Triangulation_data_structure = TDS;
-  using Vertex_handle = typename TDS::Vertex_handle;
-  using Cell_handle = typename TDS::Cell_handle;
-  using Vertex = typename TDS::Vertex;
-  using Cell = typename TDS::Cell;
-  using TDS_data = typename TDS::Cell_data;
+  using Vertex_handle = typename Cb::Vertex_handle;
+  using Cell_handle = typename Cb::Cell_handle;
+
+  using Geom_traits = Gt;
 
   // Index Type
-  using Subdomain_index = Subdomain_index_;
-  using Surface_patch_index = Surface_patch_index_;
+  using Subdomain_index = SubdomainIndex;
+  using Surface_patch_index = SurfacePatchIndex;
 
 public:
-  // Constructors
   Simplicial_mesh_cell_3()
     : time_stamp_(std::size_t(-1))
   {}
 
   Simplicial_mesh_cell_3(const Simplicial_mesh_cell_3& rhs)
-    : N(rhs.N)
-    , V(rhs.V)
-    , time_stamp_(rhs.time_stamp_)
-    , subdomain_index_(rhs.subdomain_index_)
+    : Cb(static_cast<const Cb&>(rhs)),
+      time_stamp_(rhs.time_stamp_),
+      subdomain_index_(rhs.subdomain_index_)
   {
-    for(int i=0; i <4; i++){
+    for(int i=0; i <4; ++i)
       surface_index_table_[i] = rhs.surface_index_table_[i];
-    }
-  }
-
-  Simplicial_mesh_cell_3(Vertex_handle v0,
-                         Vertex_handle v1,
-                         Vertex_handle v2,
-                         Vertex_handle v3)
-    : V(CGAL::make_array(v0, v1, v2, v3))
-  {
-  }
-
-  Simplicial_mesh_cell_3(Vertex_handle v0,
-                         Vertex_handle v1,
-                         Vertex_handle v2,
-                         Vertex_handle v3,
-                         Cell_handle n0,
-                         Cell_handle n1,
-                         Cell_handle n2,
-                         Cell_handle n3)
-    : N(CGAL::make_array(n0, n1, n2, n3))
-    , V(CGAL::make_array(v0, v1, v2, v3))
-  {
-  }
-
-  // ACCESS FUNCTIONS
-  Vertex_handle vertex(int i) const
-  {
-    CGAL_precondition( i >= 0 && i <= 3 );
-    return V[i];
   }
 
-  bool has_vertex(Vertex_handle v) const
-  {
-    return (V[0] == v) || (V[1] == v) || (V[2]== v) || (V[3]== v);
-  }
+  Simplicial_mesh_cell_3(Vertex_handle v0, Vertex_handle v1,
+                         Vertex_handle v2, Vertex_handle v3)
+    : Cb(v0, v1, v2, v3)
```

Comment:
Should these other constructors also initialize the timestamp?

Reply:
is it initialized in the `Cb` constructor?

Reply:
no, the timestamp is a member of this class (Simplicial_mesh_cell_3).

Reply:
@janetournois noticed that the initialization of the cell's timestamp to `-1` is actually performed on every new allocation of compact container new block (`initialize_time_stamp()`), so the initialization to `-1` in the cell bases is never required. Ideally it sould be cleaned from all the cell bases that do it (some in Mesh_3, Alpha_wrap_3, etc.) and maybe some comments in the code.